security/pkg/workload/server.go

```diff
@@ -86,15 +84,16 @@ func (s *SDSServer) SetServiceIdentityCert(content []byte) error {
 func (s *SDSServer) SetServiceIdentityPrivateKey(content []byte) error {
 	s.privateKeyGuard.Lock()
 	s.privateKey = content
+	s.version = fmt.Sprintf("%v", time.Now().UnixNano()/int64(time.Millisecond))
 	s.privateKeyGuard.Unlock()
 	return nil
 }
 
 // GetTLSCertificate generates the X.509 key/cert for the workload identity
 // derived from udsPath, which is where the FetchSecrets grpc request is
 // received.
-// SecretServer implementations could have diffent implementation
-func (s *SDSServer) GetTLSCertificate(udsPath string) (*auth.TlsCertificate, error) {
+// SecretServer implementations could have different implementation
+func (s *SDSServer) GetTLSCertificate() (*auth.TlsCertificate, error) {
 	s.certificateChainGuard.RLock()
```

Comment:
Do we also need to provide root cert?

Reply:
We don't need the root cert for now. Secret.TlsCertificate only requires certificate chain and private key.

https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/auth/cert.proto#L74